google-cloud-spanner/src/main/java/com/google/cloud/spanner/BuiltInOpenTelemetryMetricsProvider.java

```diff
@@ -65,8 +66,14 @@ OpenTelemetry getOrCreateOpenTelemetry(String projectId, @Nullable Credentials c
         SdkMeterProviderBuilder sdkMeterProviderBuilder = SdkMeterProvider.builder();
         BuiltInOpenTelemetryMetricsView.registerBuiltinMetrics(
             SpannerCloudMonitoringExporter.create(projectId, credentials), sdkMeterProviderBuilder);
-        this.openTelemetry =
-            OpenTelemetrySdk.builder().setMeterProvider(sdkMeterProviderBuilder.build()).build();
+        SdkMeterProvider sdkMeterProvider = sdkMeterProviderBuilder.build();
+        this.openTelemetry = OpenTelemetrySdk.builder().setMeterProvider(sdkMeterProvider).build();
+        Runtime.getRuntime()
+            .addShutdownHook(
+                new Thread(
+                    () -> {
+                      sdkMeterProvider.shutdown().join(5, TimeUnit.SECONDS);
```

Comment:
quick question: Is there any specific reason to put 5 seconds?

Reply:
Not really. I was just trying to give some upper limit for the shutdown and export to complete.

Reply:
@surbhigarg92 got it. can we use close method? close method which is provided by `SdkMeterProvider` has 10 seconds wait time. We can reuse that method.

`Runtime.getRuntime().addShutdownHook(new Thread(sdkMeterProvider::close));`

Sharing the internal implementation of `SdkMeterProvider`

```
public void close() {
    this.shutdown().join(10L, TimeUnit.SECONDS);
  }
```

Ref: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java#L166